### PR TITLE
Support jira intergration

### LINKED ISSUE #1999: Integrate Support tickets with JIRA
Currently we only recieve support tickets in slack. We need to integrate this with our JIRA platform.
A perfect solution would be an in-platform solution where we integrate JIRA directly into our support field. 

For now we might just be able to send an email from the backend to our support mailbox with the original sender as email sender. Since JIRA already polls from support@escience.sdu.dk this should fix the problem. 

Bug reports should still only send to slack, or be integrated with github (What do you think @DanThrane ).

(Issue number https://www.youtube.com/watch?v=weUhBGA8mxo )

### PR DESCRIPTION
**Pull request by hschu12**
Adds subject field to support tickets and forward tickets to support@escience.sdu.dk as well as slack.

This have not been tested since localhost gets 404 on send slack and dev does not work.
If works will fix #1999

### CODE REVIEW

File: backend/mail-service/api/src/main/kotlin/MailDescriptions.kt [hunk 1]
```diff
@@ -22,9 +22,33 @@ data class SendBulkRequest(
...
+data class SendSupportEmailRequest(
+    val fromEmail: String,
+    val subject: String,
+    val message: String
+)
+
 object MailDescriptions : CallDescriptionContainer("mail") {
     val baseContext = "/api/mail"
 
+    val sendSupport = call<SendSupportEmailRequest, Unit, CommonErrorMessage>("sendSupport") {
```
Comment (Tiggles) [inline backend/mail-service/api/src/main/kotlin/MailDescriptions.kt:34]:
```suggestion
    val sendSupport = call<SendSupportEmailRequest, SendSupportEmailResponse, CommonErrorMessage>("sendSupport") {
```

Also add `typealias SendSupportEmailResponse = Unit`.

File: backend/mail-service/src/main/kotlin/mail/services/MailService.kt [hunk 2]
```diff
@@ -106,6 +106,63 @@ class MailService(
...
+        fromEmail: String,
+        subject: String,
+        text: String
+    ) {
+        val recipientAddress = InternetAddress("support@escience.sdu.dk")
+
+        // Setup mail server
+        val properties = System.getProperties()
+        properties.setProperty("mail.smtp.host", "localhost")
+        properties.setProperty("mail.smtp.port", "25")
```
Comment (Tiggles) [inline backend/mail-service/src/main/kotlin/mail/services/MailService.kt:134]: This is done in both this function and `send`. Couldn't this be done on service initialization or is it a per request thing?

File: backend/support-service/src/test/kotlin/support/http/SupportTest.kt [hunk 3]
```diff
@@ -35,7 +35,7 @@ class SupportTest {
                     method = HttpMethod.Post,
                     path = "/api/support/ticket",
                     user = TestUsers.user,
-                    request = CreateTicketRequest("This is message")
+                    request = CreateTicketRequest("subbject", "This is message")
```
Comment (Tiggles) [inline backend/support-service/src/test/kotlin/support/http/SupportTest.kt:38]:
```suggestion
                    request = CreateTicketRequest("subject", "This is message")
```

File: frontend-web/webclient/app/ui-components/SupportBox.tsx [hunk 4]
```diff
@@ -42,11 +43,13 @@ export default function Support(): JSX.Element {
     async function onSubmit(event: React.FormEvent): Promise<void> {
         event.preventDefault();
         const text = textArea.current?.value ?? "";
+        const title = titleArea.current?.value ?? "";
         if (text.trim()) {
             try {
                 setLoading(true);
-                await Client.post("/support/ticket", {message: `${type}: ${text}`});
+                await Client.post("/support/ticket", {subject: `${title}`, message: `${type}: ${text}`});
```
Comment (Tiggles) [inline frontend-web/webclient/app/ui-components/SupportBox.tsx:50]:
```suggestion
                await Client.post("/support/ticket", {subject: title, message: `${type}: ${text}`});
```

**Review by Tiggles — COMMENTED** [review]
Looks fine. Is the subject field added necessary?

**Comment by hschu12** [comment]
The subject field is mostly for Jira. Here each ticket gets the name of the subject from the email. But no it is not strictly needed. It can be removed and we just add a title ourself.

**Review by DanThrane — APPROVED** [review]
Looks good to me. @Tiggles Should we remove the subject and just default to something else?

**Comment by Tiggles** [comment]
> Looks good to me. @Tiggles Should we remove the subject and just default to something else?

I think it's fine to have. Let's keep it for now, and remove if end-users aren't using it properly.

**Pull request merged**